lib/common/socket/uv-binding.c.h

```diff
@@ -119,26 +150,60 @@ void do_read_start(h2o_socket_t *_sock)
 {
     struct st_h2o_uv_socket_t *sock = (struct st_h2o_uv_socket_t *)_sock;
 
-    if (sock->super.ssl == NULL)
-        uv_read_start(sock->uv.stream, alloc_inbuf_tcp, on_read_tcp);
-    else
-        uv_read_start(sock->uv.stream, alloc_inbuf_ssl, on_read_ssl);
+    switch (sock->handle->type) {
+    case UV_TCP:
+        if (sock->super.ssl == NULL) {
+            uv_read_start((uv_stream_t *)sock->handle, alloc_inbuf_tcp, on_read_tcp);
+        } else {
+            uv_read_start((uv_stream_t *)sock->handle, alloc_inbuf_ssl, on_read_ssl);
+        }
+        break;
+    case UV_POLL:
+        sock->poll.events |= UV_READABLE;
+        uv_poll_start((uv_poll_t *)sock->handle, sock->poll.events, on_poll);
+        break;
+    default:
+        assert(!"FIXME");
```

Comment:
Please use `h2o_fatal`, since we would not want to see this check dropped when built with `-DNDEBUG`.

Same applies for other places.